cpp/test/Ice/invoke/AllTests.cpp

```diff
@@ -434,14 +442,63 @@ allTests(Test::TestHelper* helper)
         test(completed.get_future().get());
     }
 
+    {
+        // Call oneway with a regular (twoway) proxy
+        promise<pair<bool, vector<Ice::Byte>>> completed;
+        Ice::ByteSeq inEncaps, outEncaps;
+        cl->ice_invokeAsync(
+            "opOneway",
+            Ice::OperationMode::Normal,
+            inEncaps,
+            [&](bool ok, vector<Ice::Byte> ouEncaps)
+            {
+                completed.set_value(make_pair(ok, std::move(ouEncaps)));
+            },
+            [&](exception_ptr ex)
+            {
+                completed.set_exception(ex);
+            });
+        pair<bool, vector<Ice::Byte>> result = completed.get_future().get();
+        test(result.first);
+        test(result.second.size() == 6); // Empty encapsulation
+    }
+
     //
     // repeat with the future API.
     //
 
     {
+        // Oneway operation with a twoway proxy must return empty encapsulation.
+        Ice::ByteSeq inEncaps;
+        auto completed = cl->ice_invokeAsync("opOneway", Ice::OperationMode::Normal, inEncaps);
+        auto result = completed.get();
+        test(result.returnValue);
+        test(result.outParams.size() == 6); // Empty encapsulation
+    }
+
+    {
+        // Oneway operation with a twoway proxy must return empty encapsulation buffer.
         Ice::ByteSeq inEncaps, outEncaps;
+        auto returnValue = cl->ice_invoke("opOneway", Ice::OperationMode::Normal, inEncaps, outEncaps);
```

Comment:
That is the original issue I found when upgrading Python in main branch, the returned buffer has two bytes with values 0x0 0x0